gc3libs/utils.py

```diff
@@ -310,82 +311,6 @@ def copyfile(src, dst, overwrite=False, changed_only=True, link=False):
     return True
 
 
-def copytree(src, dst, overwrite=False, changed_only=True):
-    """
-    Recursively copy an entire directory tree rooted at `src`.
-
-    If `overwrite` is ``False`` (default), entries that already exist in
-    the destination tree are left unchanged and not overwritten.
-
-    If `overwrite` is ``True``, then `changed_only` determines
-    which files are overwritten:
-
-    - if `changed_only` is ``True`` (default), then only files for
-      which the source has a different size or has been modified
-      more recently than the destination are copied;
-
-    - if `changed_only` is ``False``, then *all* files in `source`
-      will be copied into `destination`, unconditionally.
-
-    Destination directory `dst` is created if it does not exist.
-
-    See also: `shutil.copytree`.
-    """
-    assert os.path.isdir(src), \
-        ("Source path `%s` does not name an existing directory" % src)
-    errors = []
-    if not os.path.exists(dst):
-        os.makedirs(dst)
-    for name in os.listdir(src):
-        srcname = os.path.join(src, name)
-        dstname = os.path.join(dst, name)
-        try:
-            if os.path.isdir(srcname):
-                errors.extend(
-                    copytree(srcname, dstname, overwrite, changed_only))
-            else:
-                copyfile(srcname, dstname)
-        except (IOError, os.error) as why:
-            errors.append((srcname, dstname, why))
-    return errors
-
-
-def copy_recursively(src, dst, overwrite=False, changed_only=True):
-    """
-    Copy `src` to `dst`, descending it recursively if necessary.
-
-    The `overwrite` and `changed_only` optional arguments
-    have the same effect as in `copytree`:func: (which see).
-    """
-    if os.path.isdir(src):
-        copytree(src, dst, overwrite, changed_only)
-    else:
-        copyfile(src, dst, overwrite, changed_only)
-
-
-def count(seq, predicate):
-    """
-    Return number of items in `seq` that match `predicate`.
-    Argument `predicate` should be a callable that accepts
-    one argument and returns a boolean.
-    """
-    count = 0
-    for item in seq:
-        if predicate(item):
-            count += 1
-    return count
-
-
-def defproperty(fn):
```

Comment:
See my other remark about removing `defproperty`: I'm OK with it, but `@property` is *not* a drop-in replacement for it.